force-app/main/service/AddressService.cls

```diff
@@ -151,8 +155,18 @@ public inherited sharing class AddressService {
         sobjDst.put(strFieldPrefix + 'PostalCode', addr.MailingPostalCode__c);
         sobjDst.put(strFieldPrefix + 'Latitude', addr.Geolocation__Latitude__s);
         sobjDst.put(strFieldPrefix + 'Longitude', addr.Geolocation__Longitude__s);
-        sobjDst.put(strFieldPrefix + 'State', addr.MailingState__c);
-        sobjDst.put(strFieldPrefix + 'Country', addr.MailingCountry__c);
+        String countryField = 'Country';
+        String stateField = 'State';
+        if (orgConfig.isStateCountryPicklistsEnabled()) {
+            if (addr.MailingCountry__c != null && addr.MailingCountry__c.length() <= 3) {
+                countryField = 'CountryCode';
+            }
+            if (addr.MailingState__c != null && addr.MailingState__c.length() == 2) {
```

Comment:
This is a replacement for relying on the Describe call to all State values, which lead to a bug when two countries have the same state code value... what else can we do to determine if the Address contains a Code vs a full name?